File: app/schemas/transaction.py
from datetime import datetime
from typing import List

from pydantic import BaseModel


class Base(BaseModel):
    amount: float
    installments: int
    requested_at: datetime


class Customer(BaseModel):
    avg_amount: float
    tx_count_24h: int
    known_merchants: List[str]


class Merchant(BaseModel):
    id: str
    mcc: str
    avg_amount: float


class Terminal(BaseModel):
    is_online: bool
    card_present: bool
    km_from_home: float


class LastTransaction(BaseModel):
    timestamp: datetime
    km_from_current: float


class Transaction(BaseModel):
    id: str
    transaction: Base
    customer: Customer
    merchant: Merchant
    terminal: Terminal
    last_transaction: LastTransaction

File: src/domain/transaction.rs
use chrono::{DateTime, Utc};

pub struct Transaction {
    pub id: String,
    pub amount: f32,
    pub installments: u32,
    pub requested_at: DateTime<Utc>,
    pub customer: Customer,
    pub merchant: Merchant,
    pub terminal: Terminal,
    pub last_transaction: Option<LastTransaction>,
}

pub struct Customer {
    pub avg_amount: f32,
    pub tx_count_24h: u32,
    pub known_merchants: Vec<String>,
}

pub struct Merchant {
    pub id: String,
    pub mcc: String,
    pub avg_amount: f32,
}

pub struct Terminal {
    pub is_online: bool,
    pub card_present: bool,
    pub km_from_home: f32,
}

pub struct LastTransaction {
    pub timestamp: DateTime<Utc>,
    pub km_from_current: f32,
}
